src/sync/sync_manager.hpp

```diff
@@ -20,6 +20,7 @@
 #define REALM_OS_SYNC_MANAGER_HPP
 
 #include "shared_realm.hpp"
+#include "sync/impl/sync_client.hpp"
```

Comment:
We avoid including `impl` headers from within public headers. The declaration of `ClientThreadListener` should be moved to a public header.